### DISCUSSION TITLE
Why there is no property self.trainer.is_first_batch?

### DISCUSSION THREAD

**Discussion by youngzhou1999** [Lightning-AI/pytorch-lightning#16364] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi, I'm curious about why no `is_first_batch` in `trainer`.
It's so fluent to use `self.trainer.is_last_batch` to judge the end of one epoch just as you guys post on [tutorial](https://pytorch-lightning.readthedocs.io/en/stable/model/manual_optimization.html). However, I found that there is no property `is_first_batch` in `trainer`. 
what I'm now doing is set a metric for every epoch and compute it when one epoch ends. For exampe, I want to compute the whole training data's accuracy over sth. But I'm stuck with the start point of epoches. Is ther any suggestions to get the first batch of an epoch for solving the problem? 
In my opinion, It's convenient to add the property of `self.trainer.is_first_batch` so that I can achieve what I want.

**Comment by semaphore-egg** [comment]
Maybe testing `batch_idx==0` is simple enough for this purpose.

**Reply by youngzhou1999** [reply]
ohh! It seems easy and reasonale.
